frontend/src/scenes/dashboard/DashboardTemplateVariablesLogic.ts

```diff
@@ -0,0 +1,36 @@
+import { actions, kea, path, reducers } from 'kea'
+import { DashboardTemplateVariableType, FilterType, Optional } from '~/types'
+
+import type { dashboardTemplateVariablesLogicType } from './DashboardTemplateVariablesLogicType'
+
+export const dashboardTemplateVariablesLogic = kea<dashboardTemplateVariablesLogicType>([
+    path(['scenes', 'dashboard', 'DashboardTemplateVariablesLogic']),
+    actions({
+        setVariables: (variables: DashboardTemplateVariableType[]) => ({ variables }),
+        updateVariable: (variable_name: string, filterGroup: Optional<FilterType, 'type'>) => ({
```

Comment:
would tempted to call this `setVariable` to make it clearer that `setVariables` and `setVariable` operate on the same thing differing only in number